pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -118,6 +119,77 @@ func RerunDashAlertMigration(mg *migrator.Migrator) {
 	}
 }
 
+func AddDashboardUIDPanelIDMigrations(mg *migrator.Migrator) {
+	mg.AddMigration("add dashboard_uid column to alert_rule", migrator.NewAddColumnMigration(
+		migrator.Table{Name: "alert_rule"},
+		&migrator.Column{
+			Name:     "dashboard_uid",
+			Type:     migrator.DB_NVarchar,
+			Length:   40,
+			Nullable: true,
+		},
+	))
+	mg.AddMigration("add panel_id column to alert_rule", migrator.NewAddColumnMigration(
+		migrator.Table{Name: "alert_rule"},
+		&migrator.Column{
+			Name:     "panel_id",
+			Type:     migrator.DB_BigInt,
+			Nullable: true,
+		},
+	))
+	mg.AddMigration("create index to query by dashboard_uid and panel_id", migrator.NewAddIndexMigration(
```

Comment:
nit: moving the index creation after we've populated the table should speed up the migration as we don't have to update the index on each insert.